### PR TITLE
feat(input-time-picker): support fractional seconds

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #6591 

## Summary

This PR adds initial support for fractional seconds in `input-time-picker` and `time-picker` components.  Both `input-time-picker` and `time-picker` officially support fractional step values from `.001` to `.9`.

### CODE REVIEW

File: packages/calcite-components/src/components/time-picker/assets/time-picker/t9n/messages.json [hunk 1]
```diff
@@ -5,6 +5,9 @@
   "meridiem": "AM/PM",
   "meridiemDown": "Decrease AM/PM",
   "meridiemUp": "Increase AM/PM",
+  "fractionalSecond": "Fractional second",
```
Comment (jcfranco) [inline packages/calcite-components/src/components/time-picker/assets/time-picker/t9n/messages.json:8]: Can you also update `messages_en.json`? Both need to be in sync. cc @anveshmekala

**Review by jcfranco — APPROVED** [review]
Awesome stuff, @eriklharper! ✨🕐✨

The critical things to address before merging are:

- [x] bound functions used in render functions
- [x] explicit `tabindex` usage to receive focus in time picker parts
- [x] add story or stories to cover in screenshot testing
- [x] updating `messages_en.json`
- [x] math util cleanup (if possible)
- [x] add usage file for milliseconds

Anything cleanup related can be tackled in a separate PR.

**Comment by jcfranco** [comment]
@eriklharper Before I forget, can you add more info about this feature in the description and also create follow-up issues for under the `TO DO` section?

>  Should max end of range be subtracted by the step when nudging down from the minimum value? (this is what the native time input does in Chrome)

Do you have an example of what this looks like? I'm afraid I'm not quite following.

**Comment by jcfranco** [comment]
One more thing 😅, could you also add a usage file for this configuration?

File: packages/calcite-components/src/components/time-picker/time-picker.tsx [hunk 2]
```diff
@@ -898,7 +1000,54 @@ export class TimePicker
...
+        )}
+        {this.showFractionalSecond && (
+          <div class={CSS.column} role="group">
+            <span
+              aria-label={this.messages.fractionalSecondUp}
+              class={{
+                [CSS.button]: true,
+                [CSS.buttonFractionalSecondUp]: true,
+              }}
+              onClick={this.nudgeFractionalSecond.bind(this, "up")}
```
[the remark this answers is not in the document]
Comment (eriklharper) [reply packages/calcite-components/src/components/time-picker/time-picker.tsx]: I just added increment and decrement functions that both call the new nudge function.  That feels cleaner than the data attribute method.

File: packages/calcite-components/src/components/time-picker/resources.ts [hunk 3]
```diff
@@ -2,6 +2,8 @@ export const CSS = {
   button: "button",
   buttonBottomLeft: "button--bottom-left",
   buttonBottomRight: "button--bottom-right",
+  buttonFractionalSecondDown: "button--fractionalSecond-down",
```
[the remark this answers is not in the document]
Comment (eriklharper) [reply packages/calcite-components/src/components/time-picker/resources.ts:5]: Not following.  The pattern here is the same as the pattern elsewhere in the file for other items.

**Pull request merged**